baremetal/metal3machine_manager_test.go

```diff
@@ -1397,6 +1397,12 @@ var _ = Describe("Metal3Machine manager", func() {
 					expectedName = tc.ExpectedConsumerRef.Name
 				}
 				Expect(name).To(Equal(expectedName))
+				if machineMgr.Metal3Machine.Status.MetaData == nil {
```

Comment:
None of the bmh you are working with in the tests have the networkData and metaData set I think, so you should modify bmhSpecSomeImg for example to add it there, so that it is actually tested